code/game/objects/effects/particle_holder.dm

```diff
@@ -5,63 +5,52 @@
 	mouse_opacity = MOUSE_OPACITY_TRANSPARENT
 	layer = ABOVE_ALL_MOB_LAYER
 	vis_flags = VIS_INHERIT_PLANE
-	///typepath of the last location we're in, if it's different when moved then we need to update vis contents
-	var/last_attached_location_type
-	///the main item we're attached to at the moment, particle holders hold particles for something
-	var/datum/weakref/weak_attached
-	///besides the item we're also sometimes attached to other stuff! (items held emitting particles on a mob)
-	var/datum/weakref/weak_additional
+	/// Holds info about how this particle emitter works
+	/// See \code\__DEFINES\particles.dm
+	var/particle_flags = NONE
 
-/obj/effect/abstract/particle_holder/Initialize(mapload, particle_path = /particles/smoke)
+/obj/effect/abstract/particle_holder/Initialize(mapload, particle_path = /particles/smoke, particle_flags = NONE)
 	. = ..()
 	if(!loc)
 		stack_trace("particle holder was created with no loc!")
 		return INITIALIZE_HINT_QDEL
-	if(ismovable(loc))
-		RegisterSignal(loc, COMSIG_MOVABLE_MOVED, PROC_REF(on_move))
-	RegisterSignal(loc, COMSIG_PARENT_QDELETING, PROC_REF(on_qdel))
-	weak_attached = WEAKREF(loc)
+	// We assert this isn't an /area
+
+	src.particle_flags = particle_flags
 	particles = new particle_path
-	update_visual_contents(loc)
+	// /atom doesn't have vis_contents, /turf and /atom/movable do
+	var/atom/movable/lie_about_areas = loc
+	lie_about_areas.vis_contents += src
+	if(!ismovable(loc))
+		RegisterSignal(loc, COMSIG_PARENT_QDELETING, PROC_REF(immovable_deleted))
+
+	if(particle_flags & PARTICLE_ATTACH_MOB)
+		RegisterSignal(loc, COMSIG_MOVABLE_MOVED, PROC_REF(on_move))
+	on_move(loc, null, NORTH)
```

Comment:
could be a nothing burger but i think this would be easier to "get" if args were named here

Reply:
uh, sure